### ISSUE TITLE
Rewrite GUI

### ISSUE THREAD

**Issue by NicTanghe** [moneymanagerex/moneymanagerex#2687]
# The GUI uses a dated framework

People are looking at remaking it.
Currently considered is

- Avalonia a .NET core framework

**Comment by thinkpad4** [comment]
I like the current look, too many things are flat and boring, at least this has BUTTONS not flat nonsense

**Comment by NicTanghe** [comment]
Hey if you don't like millions of minds design conclusions about efficient
communicative stress reducing design then beleav whatever u want.

The buttons are indeed not that bad tho.
The main uglyness in the design is that it doesnt have one. UX is just
completly ignored and estithics are also largly ignored like most apps
created by pre 2010 graduated developers

**Comment by vomikan** [comment]
@NicTanghe 
You're right. But you probably don't understand how this application is made.
The library is taken as a basis from 1992.
https://en.wikipedia.org/wiki/WxWidgets
The design is limited to the tools presented in it.

**Comment by NicTanghe** [comment]
So just redo the ui by using a different library? Or is the the gui 2
intertwined with the functionality. If that is the case and i whouldnt have
just started a new study i whould have tried to jumpstart by building a gui
on flutter. But i dont have enough free time now.

**Comment by gabriele-v** [comment]
@NicTanghe have you ever used Filezilla? 
It's based on the same UI components (wxWidgets): is it ugly and feels old? Yes, probably.
But I think no one complains about FileZilla UI because simply it's working (and working well to be honest).

**Comment by NicTanghe** [comment]
Or becouse only old people are using it and they prefer outdated design
choises?

Becouse that feels more familiar to them even tho it is objectivly worse ?

If you want simple and functional just use a terminal app it probably even
looks better then this becouse you can use a logical good looking
colourscheme.

**Comment by thinkpad4** [comment]
@NicTanghe Why does everything that works well need to be "Modernized"?  Just about every program and website that "Modernized" got worse and worse to the point of unusability.

**Comment by NicTanghe** [comment]
Those got "modernised" by people that diden't propperly understand ux and only understood UI.

Oftentimes they only hired a designer to redesign everything and designers verry verry often dont have knowledge of ux. ANd its cheaoer ti hire a good designer then to higher someone with good ux knowledge.
It could also just be that the old design rusted in place in your brain and everything diferent then what rusted in place is experianced by the user. dealing with these longtime users is always hard.
My filosofy is keep al bindings(Unless the some bidings have retarded defaults) and change in 3 patches while stating the final design from the start.

**Comment by NicTanghe** [comment]
Ps paper worked very well so why the fuck are u using a computer.

**Comment by stef145g** [comment]
Hi @NicTanghe your argument is very emotional rather than factual.

As this is an open source project, everyone has the capacity to contribute to the project to make it better. You say that you could design a better UI, then please show some examples and give some constructive comments/criticisms rather than just denigrating the efforts of past and present designers that have created and maintained this project over many years. 

Your comments appear to be a springboard for advertising and provide no positive direction or input. As an "old person" and not so active contributor these days, I find your comments very offensive.

**Comment by NicTanghe** [comment]
> 
> 
> Hi @NicTanghe your argument is very emotional rather than factual.
> 
> As this is an open source project, everyone has the capacity to contribute to the project to make it better. You say that you could design a better UI, then please show some examples and give some constructive comments/criticisms rather than just denigrating the efforts of past and present designers that have created and maintained this project over many years.
> 
> Your comments appear to be a springboard for advertising and provide no positive direction or input. As an "old person" and not so active contributor these days, I find your comments very offensive.

Wel my comment is not emotional and it is at the same time.
UX and UI is 50% about how it feels to use.

What is happening is i was trying to figure out if i could help.
After i heard the project is using a 30 year old framework i was starting to pres hard on the fact things needed to change "upstream".

I got a verry emotional comment in the sence that i am used to this and as an individual i prefer it and it works. 

Then i tried to illustrate what a ridicules argument that is by stating paper worked and allot of people prefer it so why are you using a computer.

Just becouse someone uses your age as an argument doesnt mean the argument is inavalid. I got paid a metric ton les then older people because of inexperience but now that I'm older i c why it is important to have experience and confidence.

i was trying to get a conversation started and the next step in this conversation was again asking for a reply on how much of the functionality is relyant on this 30 year old framework and how much could be replaced by flutter.

** TLDR **
---

- I need to know if I spend time making a modern ui that it can work.

- using a 30 year old framework is often not a verry good idea whem it has bad design principals.

- buy a mirror and look at your emotions.

**Comment by NicTanghe** [comment]
If you want to use c++ u should probably w8 until winUI 3 releases annyway.

It will probably be a solid framework when that version drops.

**Comment by NicTanghe** [comment]
Seems People are on it.
I whould like to help but im not a erry good coder.
I do get classes this year but its more focused on the vfx world.

Learning .net and avalonia is something i have no time for.
So I'm afraid i won't be of assistance.

maybe Qt and C++ is an option ?

**Comment by MartinX3** [comment]
@NicTanghe the current wxWidgets version is 3.1.4 from 2020-07-22
3.2 is in the works as well to cut old code away.

I think upgrading mmex to use the newest wxWidgets and refactor the mmex UI code on top of it is faster than changing the entire lib and rewrite the entire UI.
Especially since wxWidgets is still being developed.

If I looked correctly mmex currently uses 2.9.2 from 2011-07-05.

**Comment by vomikan** [comment]
Version: 1.3.6-RC.1
• Database version: 7 • (aes128cbc)
MMEX is using the following support products:
• wxWidgets 3.1.4 (wxMSW 6.2)
• wxSQLite3 4.6.0 (SQLite 3.31.1)
• RapidJSON 1.1.0
• Lua 5.3.5
• libcurl/7.71.1 Schannel
Build on Aug 12 2020 08:12:25 with:
• CMake 3.16.2
• MSBuild 15.9.21.664
• gettext 0.19.8.1
• Microsoft Visual Studio 15.0
• Microsoft Visual C++ 19.16.27035.0
• Windows SDK 10.0.14393.0
Running on:
• Windows 8 (build 9200), 64-bit edition
• (windows-1251)
• 1920x1080 32bit 96x96ppi

**Comment by NicTanghe** [comment]
> wxWidgets

Dear person dude.

When I say the lib is outdated I don't mean want updated recently I mean as far as I know if you use it your program will look Outdated and Ugly. And since the only thing that needs to be rewritten is the GUI and not the core functionality of the program I Firmly beleave a complete rework of the GUI part of the code is in order.

I do note that my current knowledge on wxWidgets is very limited but it seems to always be real ugly no matter what if it's used on Windows.

I have yet to find any program using it that doesn't look like its some pet project garbage someone made early 2000s by someone who does not understand that looking good is important. This programme deserves better than looking like that.

**Comment by thinkpad4** [comment]
@vomikan Hang on, you are running Version: 1.3.6-RC.1 but the latest on the releases page is Version: 1.3.5.   The only place I see Version: 1.3.6-RC.1 is under nightly but that says May 6th and the stable says June 15th.

**Comment by savornicesei** [comment]
@NicTanghe and @all,

Before deciding on how you should decide on what - what should the modern UI of mmex look like? As a user I would like to see some designs and mockups of the new UI. Since the mmex project is not so known, asking for help on UI/UX on reddit might be a solution to get the needed input and help.

In regards to the UI libraries to be used, using Avalonia.NET will require .NET SDK which is not available in official repos. 
Since you also have an android app, KDE libs can cover all your UI with proper scaling on mobile and nice, modern UI on win, linux and android.

**Comment by NicTanghe** [comment]
@savornicesei  i`m also just a user for now as i have to spend all my time on school,
I whould rather not have this project use a .net library and only mentioned it because someone else did.

I`d suggest QT since its more usefull to learn then kde.

**Comment by gabriele-v** [comment]
Probably you don't know what should be done to move it to QT.

As I said for me it's completely useless, if we need to change the GUI, we have to move to .NET that it's the future. Which part of the actual C++ code you could keep moving to another GUI? Probably just the ORM, that with .NET libraries comes already "for free", so just my 2 cents it doesn't make any sense for me.

**Comment by NicTanghe** [comment]
I don't know in what reality a Windows run language is the future but I don't think it's mine. Unless I'm mistaken and it's not Windows run.

**Comment by vomikan** [comment]
I just read this article
https://habr.com/ru/post/143911/
The author paid attention to many aspects such as development time. Time is what we lack.

![image](https://user-images.githubusercontent.com/6836805/102685581-e8e2c980-41f2-11eb-8da0-397d32bd7ca7.png)


To arrange holivar and endless discussions — it is also a fad of some people in almost any community

**Comment by NicTanghe** [comment]
IIn my opinion you should stay away from .Microsoft
